authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/broker/services/api/CatalogApiService.kt

```diff
@@ -83,9 +84,12 @@ class CatalogApiService(
     }
 
     private fun buildCatalogDataOffer(dataOfferRs: DataOfferListEntryRs): CatalogDataOffer {
+        val asset = dataOfferMapper.readUiAsset(dataOfferRs.assetUiJson)
+
         return CatalogDataOffer(
             assetId = dataOfferRs.assetId,
             assetTitle = dataOfferRs.assetTitle,
+            assetDataSourceAvailability = asset.dataSourceAvailability,
```

Comment:
Consequence of the above: You can skip reading the asset and return the availability directly